packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx

```diff
@@ -36,28 +37,22 @@ import {
 import { useMessageRelay } from "../MessageRelayContext";
 import { Waiting } from "./Waiting";
 
-function mapEventCategoryToBackgroundColor(eventCategory: string): string {
-	switch (eventCategory) {
-		case "generic":
-			return tokens.colorPaletteGreenForeground1;
-		case "performance":
-			return tokens.colorPaletteBlueForeground2;
-		case "error":
-			return tokens.colorPaletteRedBackground3;
-		default:
-			return tokens.colorNeutralBackground1;
-	}
-}
-
 /**
  * Set the default displayed size to 100.
  */
 const DEFAULT_PAGE_SIZE = 100;
 
+interface TelemetryViewProps {
+	/**
+	 * Object representing the current page theme.
+	 */
+	theme: { name: string; theme: Theme };
```

Comment:
Rather than plumbing our theme selection everywhere that needs it, it might be better to setup a React Context for the theme data. That way any component that needs it can grab it from the context, rather than requiring us to pass it down explicitly everywhere. 

This is the pattern we use for the MessageRelay for sending/receiving messages from the webpage.

Reply:
And we do have that "wrapper" Rohan put in place for the theme, right? Not sure if it gives us a way to get the name or its only function is to apply the theme to all child content.